contrib/generic_proxy/filters/network/source/stats.h

```diff
@@ -2,24 +2,50 @@
 
 #include <string>
 
+#include "envoy/server/factory_context.h"
+#include "envoy/singleton/instance.h"
 #include "envoy/stats/scope.h"
 #include "envoy/stats/stats_macros.h"
 
+#include "source/common/stats/symbol_table.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace NetworkFilters {
 namespace GenericProxy {
 
 /**
- * All generic filter stats. @see stats_macros.h
+ * All code and flags that generic proxy filter can emit. This should be global singleton.
+ */
+class CodeOrFlags : public Singleton::Instance {
```

Comment:
or if possible we can hide this singleton into GenericFilterStatsHelper

Reply:
> feels like you should move this to the interface/, or add an abstraction class.

> Since it is wired you reference it in the interface

I am not sure. In fact there is no strict limitation that the code in the `interface/` could not reference the code in the `source/`.

The `GenericProxyStats` and `AccessLog` have been referenced in previous implementation.

And we are going to move part of header file (route.h proxy_config.h codec_callback.h) from `interface/` to `source/`.

Only the interface that need third party developers to implement will be kept in the `interface/`.

Reply:
I see, thanks